src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/InitializeStaticFieldsInlineTests.cs

```diff
@@ -370,6 +370,41 @@ static C2()
             }.RunAsync();
         }
 
+        [Fact, WorkItem(5784, "https://github.com/dotnet/roslyn-analyzers/issues/5784")]
+        public async Task CA1810_InitializeMemberOfAnotherClass()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+public class C1
+{
+    internal static int Size = 0;
+}
+
+public class C2
+{
+    static C2()
+    {
+        C1.Size = 42;
```

Comment:
What would be the result if the static ctor initializes both static fields of another type and also static fields of the containing type? Would it match our expected result in that case?